### PR TITLE
Feature/kitosudv 2357 task 2433

### PR DESCRIPTION
**Pull request by mrjsawdk**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [x] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [x] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/DataProcessingRegistrationWriteModelMapper.cs [hunk 1]
```diff
@@ -61,46 +67,102 @@ private IEnumerable<UpdatedExternalReferenceProperties> MapReferences(IEnumerabl
...
+                    rule.MustUpdate(x => x.General.InsecureCountriesSubjectToDataTransferUuids)
+                        ? dto.InsecureCountriesSubjectToDataTransferUuids.FromNullable().AsChangedValue()
+                        : OptionalValueChange<Maybe<IEnumerable<Guid>>>.None,
+
+                DataProcessorUuids = rule.MustUpdate(x => x.General.DataProcessorUuids)
+                    ? dto.DataProcessorUuids.FromNullable().AsChangedValue()
+                    : OptionalValueChange<Maybe<IEnumerable<Guid>>>.None,
+
+                HasSubDataProcessors = rule.MustUpdate(x => x.General.HasSubDataProcessors)
+                    ? (dto.HasSubDataProcessors?.ToYesNoUndecidedOption()).AsChangedValue()
```
Comment (JacobMalling) [inline Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/DataProcessingRegistrationWriteModelMapper.cs:113]: Bliver denne nulstillet hvis man sender "Null" som value her?

Comment (JacobMalling) [reply Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/DataProcessingRegistrationWriteModelMapper.cs:113]: Ved mig skal jeg lave ?? operator for at null bliver set som en ændring

File: Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/DataProcessingRegistrationWriteModelMapper.cs [hunk 2]
```diff
@@ -61,46 +67,102 @@ private IEnumerable<UpdatedExternalReferenceProperties> MapReferences(IEnumerabl
...
+                OversightOptionUuids = rule.MustUpdate(x => x.Oversight.OversightOptionUuids)
+                    ? dto.OversightOptionUuids.FromNullable().AsChangedValue()
+                    : OptionalValueChange<Maybe<IEnumerable<Guid>>>.None,
+
+                OversightOptionsRemark = rule.MustUpdate(x => x.Oversight.OversightOptionsRemark)
+                    ? dto.OversightOptionsRemark.AsChangedValue()
+                    : OptionalValueChange<string>.None,
+
+                OversightInterval = rule.MustUpdate(x => x.Oversight.OversightInterval)
+                    ? (dto.OversightInterval?.ToIntervalOption()).AsChangedValue()
```
Comment (JacobMalling) [inline Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/DataProcessingRegistrationWriteModelMapper.cs:137]: Bliver denne nulstillet hvis man sender "Null" som value her?

**Review by JacobMalling — APPROVED** [review]
Nogle få ting der lige skal tjekkes efter. Nok også med tests der verificerer det

File: Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/DataProcessingRegistrationWriteModelMapper.cs (hunk 1 above)
Comment (mrjsawdk) [reply Presentation.Web/Controllers/API/V2/External/DataProcessingRegistrations/Mapping/DataProcessingRegistrationWriteModelMapper.cs:113]: Virker fordi det er nullable value type som target

**Pull request merged**